### PR TITLE
Removal of cuda hardcoded string with get_device function

### PR DESCRIPTION
**Pull request by raza-sikander**
In UTs removed 'cuda' string hardcode by replacing with device variable set to get_accelerator().device_name()

### CODE REVIEW

**Comment by raza-sikander** [comment]
> @raza-sikander please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Intel Corporation"

File: tests/unit/hybrid_engine/test_he_lora.py [hunk 1]
```diff
@@ -125,7 +126,8 @@ def get_model(self, model_name):
         model_config.dropout = 0.0
         model = AutoModelForCausalLM.from_pretrained(model_name, config=model_config)
         model = model.half()
-        model = model.to(f'cuda:{local_rank}')
+        device = get_accelerator().device_name()
```
Comment (loadams) [inline tests/unit/hybrid_engine/test_he_lora.py:129]: I'm not sure the hybrid engine currently supports anything other than Cuda and only certain devices at the moment, @lekurile to confirm?

File: tests/unit/ops/transformer/inference/test_softmax.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def run_softmax_ds(input, use_triton_ops=False):
 def test_softmax(batch, sequence, channels, dtype, use_triton_ops):
     if not deepspeed.HAS_TRITON and use_triton_ops:
         pytest.skip("triton has to be installed for the test")
-    input_ds = torch.randn((batch, sequence, channels), dtype=dtype, device='cuda')
```
Comment (loadams) [inline tests/unit/ops/transformer/inference/test_softmax.py:46]: @raza-sikander - does this mean we can add any additional test coverage for Gaudi2?

File: tests/unit/hybrid_engine/test_he_lora.py (hunk 1 above)
Comment (nelyahu) [reply tests/unit/hybrid_engine/test_he_lora.py:129]: @loadams We added support for Hybrid Engine via our fork, using the existing fallback python implementation to the cuda kernels within the injected layers. 
We successfuly able to train deepspeed-chat step3 on llamav2-7b using HE.
We are in the process of PRing it - as it requires some cosmetics and generalization.

**Pull request merged**